src/common/os/posix/os_utils.cpp

```diff
@@ -393,4 +395,147 @@ void CtrlCHandler::handler(void*)
 	terminated = true;
 }
 
+namespace {
+
+	class Converter
+	{
+	public:
+		virtual char* convert(const char* fromBuf, size_t fromLen) = 0;
+		virtual void destroy(char* &outBuf) = 0;
+	};
+
+	// No real conversion, system locale is UTF-8 or iconv is not available
+	class NullConverter : public Converter
+	{
+		char* convert(const char* fromBuf, size_t fromLen) override
+		{
+			return const_cast<char*>(fromBuf);
+		}
+		void destroy(char* &outBuf)
+		{
+			outBuf = NULL;
+		}
+	};
+
+#ifdef HAVE_ICONV_H
+	class IConverter : public Converter
+	{
+	public:
+
+		IConverter(const char* sysLocale) : ic(NULL)
```

Comment:
Re posix port. I do not see big problems with having incomplete OS-A build provided author is using OS-B. This can be fixed a bit later (specially at pre-alpha stage). Certainly it does not mean that code for OS-A may get poisoned with bad programming style.